From f8ec2df71bfe6a88ec6e602c6690d4da332dac2c Mon Sep 17 00:00:00 2001
From: hule <hule@amperecomputing.com>
Date: Sun, 7 Nov 2021 22:11:10 +0700
Subject: [PATCH] Fix press Reset button incorrect behavior

Pressing RESET button has the BMC incorrectly executes power cycle.
Fix the issue by changing to trigger hard reset instead

Signed-off-by: hule <hule@amperecomputing.com>
---
 src/button_handler.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/button_handler.cpp b/src/button_handler.cpp
index 6731e0f..a17b0a3 100644
--- a/src/button_handler.cpp
+++ b/src/button_handler.cpp
@@ -199,7 +199,7 @@ void Handler::resetPressed(sdbusplus::message::message& msg)
         log<level::INFO>("Handling reset button press");
 
         std::variant<std::string> state =
-            convertForMessage(Host::Transition::Reboot);
+            convertForMessage(Host::Transition::ForceWarmReboot);
 
         auto service = getService(HOST_STATE_OBJECT_NAME, hostIface);
         auto method = bus.new_method_call(
-- 
2.17.1

